File: Leetcode_Notes/面试题_16.11.跳水板/面试题_16.11.跳水板.cpp
class Solution {
public:
    vector<int> divingBoard(int shorter, int longer, int k) {
        vector<int> result;//存储所有值
        set<int> value;//存关键字（长度）
        if(k==0)
        {
            return result;
        }
        else if(shorter==longer)
        {
            int length=shorter*k;
            result.push_back(length);
            return result;
        }
        for(int i=0;i<=k;i++)
        {
            int length=(k-i)*shorter+i*longer;
            if(value.count(length)==0)
            {
                value.insert(length);
                result.push_back(length); 
            }
        }
        return result;
    }
};

File: Leetcode_Notes/面试题_16.11.跳水板/面试题_16.11.跳水板.py
def divingBoard(shorter,longer,k):
    value=set()
    result=[]
    if k==0:
        return []
    elif shorter==longer:
        return [shorter*k]
    for i in range(k+1):
        if((i*longer+(k-i)*shorter) not in value):
            value.add(i*longer+(k-i)*shorter)
            result.append(i*longer+(k-i)*shorter)
    #result=list(value)
    return result

l=divingBoard(1,2,0)
print(l)
l=divingBoard(2,2,2)
print(l)
l=divingBoard(2,168,56)
print(l)